pkg/cloud/cloud.go

```diff
@@ -521,8 +522,14 @@ func (c *cloud) CreateSnapshot(ctx context.Context, volumeID string, snapshotOpt
 
 	var tags []*ec2.Tag
 	for key, value := range snapshotOptions.Tags {
-		tags = append(tags, &ec2.Tag{Key: &key, Value: &value})
+		copiedKey := key
+		copiedValue := value
+		tags = append(tags, &ec2.Tag{Key: &copiedKey, Value: &copiedValue})
 	}
+	// Enforce ordering for unit tests.
+	sort.SliceStable(tags, func(i, j int) bool {
+		return *tags[i].Key < *tags[j].Key
+	})
```

Comment:
Is this sorting really necessary? If so, can we potentially move it to the test function?

Reply:
Removed in favor of a custom matcher for this AWS input type. Let me know if this is what you were thinking.